frontend/model/contracts/chatroom.js

```diff
@@ -393,6 +399,15 @@ sbp('chelonia/defineContract', {
         if (msgIndex >= 0) {
           state.messages.splice(msgIndex, 1)
         }
+        // filter replied messages and check if the current message is original
+        for (const message of state.messages) {
+          if (message.replyingMessage) {
+            if (message.replyingMessage.id === data.id) {
```

Comment:
No big deal but these two conditions could be merged into one with `&&`

Reply:
if (message.replyingMessage?.id === data.id)
And this is better solution, isn't it?

Reply:
Indeed (provided `data.id` cannot be undefined)

Reply:
Right. Only `message.replyingMessage` could be undefined.
```
if (message.replyingMessage?.id === data.id)
```